docassemble/AssemblyLine/data/questions/al_visual.yml

```diff
@@ -66,3 +78,224 @@ image sets:
+      copyText.select();
+      copyText.setSelectionRange(0, 99999);
+      document.execCommand("copy");
+
+      var tooltip = document.getElementById("myTooltip");
+      tooltip.innerHTML = "Copied: " + copyText.value;
+    }
+
+    function outFunc() {
+      var tooltip = document.getElementById("myTooltip");
+      tooltip.innerHTML = "Copy to clipboard";
+    }
+  </script>
+css: |
+  <style>
+  .copy-tooltip {
+    position: relative;
+    display: inline-block;
+  }
+
+  .copy-tooltip .tooltiptext {
+    visibility: hidden;
+    width: 140px;
+    background-color: #555;
+    color: #fff;
+    text-align: center;
+    border-radius: 6px;
+    padding: 5px;
+    position: absolute;
+    z-index: 1;
+    bottom: 150%;
+    left: 50%;
+    margin-left: -75px;
+    opacity: 0;
+    transition: opacity 0.3s;
+  }
+
+  .copy-tooltip .tooltiptext::after {
+    content: "";
+    position: absolute;
+    top: 100%;
+    left: 50%;
+    margin-left: -5px;
+    border-width: 5px;
+    border-style: solid;
+    border-color: #555 transparent transparent transparent;
+  }
+
+  .copy-tooltip:hover .tooltiptext {
+    visibility: visible;
+    opacity: 1;
+  }
+  </style>
+back button label: |
+  Back to your form
+continue button label: |
+  Send by text or email  
+---
+code: |
+  def is_phone_or_email(text):
+    if re.match("\S+@\S+", text) or phone_number_is_valid(text):
+      return True
+    else:
+      validation_error("Enter a valid phone number or email address")
+---
+id: Results of sharing
+continue button field: al_share_results
+question: |
+  Thanks for sharing!
+---
+need:
+  - al_share_answers_message_template
+  - al_tell_a_friend_message_template
+code: |
+  if phone_number_is_valid(share_interview_contact_method):
+    if sharing_type == "tell_friend":
+      success = send_sms(to=share_interview_contact_method, template=al_tell_a_friend_message_template)
+    else:
+      success = send_sms(to=share_interview_contact_method, template=al_share_answers_message_template)
+  else:
+    if sharing_type == "tell_friend":
+      success = send_email(to=share_interview_contact_method, template=al_tell_a_friend_message_template)
+    else:
+      success = send_email(to=share_interview_contact_method, template=al_share_answers_message_template)
+  al_did_share_form = True
+---
+template: al_share_answers_message_template
+subject: |
+  CourtFormsOnline form from ${ al_share_form_from_name }
+content: |
+  ${ share_interview_answers_message }
+  Click the link below to view and edit ${ al_share_form_from_name }'s
+  progress so far:
+  
+  ${ interview_url(temporary=48) }
+---
+template: al_tell_a_friend_message_template
+subject: |
+  ${ al_share_form_from_name } wants to tell you about Court Forms Online
+content: |
+  ${ tell_a_friend_message }
+  ${ interview_url(i=user_info().filename, style="short", new_session=1) }
+---
+objects:
+  - github_readonly: DAWeb.using(base_url="https://api.github.com")
```

Comment:
Something to keep in mind: the rate limit for unauthenticated github api calls is 60 an hour. We could easily go over that in prod. Should probably get the github auth token and pass it along. In curl, it's something like should just be the `Authorization: <base64encoded "user:password" string>` header, not sure what DAWeb has.

Reply:
We already have a token for the feedback form so I will change this API call to make use of it.